File: Interpreted/mutual-recursion/mrec.rb
def F(n)
  n == 0 ? 1 : n - M(F(n-1))
end
def M(n)
  n == 0 ? 0 : n - F(M(n-1))
end

puts F(ARGV[0].to_i);
puts M(ARGV[0].to_i);

File: Interpreted/mutual-recursion/mrec.exs
defmodule MutualRecursion do
  def f(0), do: 1
  def f(n), do: n - m(f(n - 1))
  def m(0), do: 0
  def m(n), do: n - f(m(n - 1))
end

[n | _] = System.argv
{n, _} = Integer.parse(n)

IO.puts MutualRecursion.f(n)
IO.puts MutualRecursion.m(n)
